beluga/include/beluga/views/forward.hpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_VIEWS_FORWARD_HPP
+#define BELUGA_VIEWS_FORWARD_HPP
+
+#include <range/v3/view/adaptor.hpp>
+
+#include <beluga/detail/prologue.hpp>
+
+/**
+ * \file
+ * \brief Implementation of a forward range adaptor object.
+ */
+
+namespace beluga::views {
+
+namespace detail {
+
+/// Implementation of a forward view adaptor.
+/**
+ * Restricts a range to model the forward range concept or lower.
+ */
+template <class Range>
+struct forward_view : public ranges::view_adaptor<forward_view<Range>, Range, ranges::range_cardinality<Range>::value> {
+ public:
+  /// Default constructor.
+  forward_view() = default;
+
+  /// Construct the view from an existing range.
+  constexpr explicit forward_view(Range range) : forward_view::view_adaptor{std::move(range)} {}
```

Comment:
Codecov says that this line is not covered, probably because it is optimized away. :shrug:

https://app.codecov.io/gh/Ekumen-OS/beluga/pull/282/blob/beluga/include/beluga/views/forward.hpp#L42